File: Python/programmers/etc/level1/콜라츠_추측.py
def solution(num):
    cnt = 0
    while num != 1:
        if ~num&1:
            num >>= 1
        else:
            num = num*3+1
        cnt += 1
        if cnt == 500:
            return -1
    return cnt


num = 6
print(solution(num)) # 8

num = 16
print(solution(num)) # 4

num = 626331
print(solution(num)) # -1

File: Java/programmers/etc/level1/콜라츠추측.java
package CodeTest.Java.programmers.etc.level1;

public class 콜라츠추측 {
    public int solution(long num) {
        int answer = 0;

        while(num != 1) {
            answer++;
            if (answer==500) return -1;
            num = (num%2==0) ? num/2 : num*3+1;
        }
        
        return answer;
    }
}
